tool/truffle/jruby_truffle_runner/lib/jruby_truffle_runner.rb

```diff
@@ -0,0 +1,285 @@
+    end
+
+    subcommand = subcommand.to_sym
+
+    subcommand_option_parser = @option_parsers[subcommand] || raise("unknown subcommand: #{subcommand}")
+    argv_after_subcommand    = subcommand_option_parser.order argv_after_global
+
+    print_options
+    help subcommand if @options[subcommand][:help]
+
+    send "subcommand_#{subcommand}", argv_after_subcommand
+  end
+
+  def print_options
+    if verbose?
+      puts 'Options:'
+      pp @options
+    end
+  end
+
+  private
+
+  def verbose?
+    @options[:global][:verbose]
+  end
+
+  def build_option_parsers
+    @option_parsers = OPTION_DEFINITIONS.each_with_object({}) do |(group, group_options), parsers|
+      parsers[group] = OptionParser.new do |option_parser|
+        group_options.each do |option, data|
+          *args, description, block, default = data
+
+          option_parser.on(*args, description + " (default: #{default.inspect})") do |new_value|
+            old_value               = @options[group][option]
+            @options[group][option] = block.call new_value, old_value
+          end
+        end
+      end
+    end
+
+    @option_parsers.each { |key, option_parser| option_parser.banner = HELP[key] }
+  end
+
+  def load_local_yaml_configuration
+    yaml_path = File.join Dir.pwd, '.truffle.yaml'
+
+    unless File.exist? yaml_path
+      candidates = Dir['*.gemspec']
+      if candidates.size == 1
+        gem_name, _ = candidates.first.split('.')
+
+        default_configuration_file_path = File.dirname(__FILE__) + "/../gem_configurations/#{gem_name}.yaml"
+        if File.exist?(default_configuration_file_path)
+          puts "Copying default .truffle.yaml for #{gem_name}."
+          FileUtils.cp default_configuration_file_path, '.truffle.yaml'
+        end
+      end
+    end
+
+    yaml_data = YAML.load_file(yaml_path) if File.exist?(yaml_path)
+    @options  = deep_merge @options, yaml_data
+  end
+
+  def construct_default_options
+    @options = OPTION_DEFINITIONS.each_with_object({}) do |(group, group_options), options|
+      group_options.each_with_object(options[group] = {}) do |(option, data), group_option_defaults|
+        *args, block, default         = data
+        group_option_defaults[option] = default
+      end
+    end
+  end
+
+  def help(key = nil)
+    parsers = if key
+                [@option_parsers[key]]
+              else
+                @option_parsers.values
+              end
+    puts *parsers
+    exit
+  end
+
+  def deep_merge(a, b)
+    if Hash === a
+      if Hash === b
+        return a.merge(b) { |k, ov, nv| deep_merge ov, nv }
+      else
+        return a
+      end
+    end
+
+    if Array === a
+      if Array === b
+        return a + b.map { |v| eval_yaml_strings v }
+      else
+        return a
+      end
+    end
+
+    eval_yaml_strings b
+  end
+
+  def eval_yaml_strings(value)
+    if String === value
+      begin
+        eval('"' + value.gsub(/\\#|"|\\/, '\#' => '\#', '"' => '\"', '\\' => '\\\\') + '"')
```

Comment:
It has to be first directory in `$:`, if t's just added with `-I` then it ends up at the end of `$:`. But it could be default that the `bundler/setup.rb` adds the path always without extra configuration in yaml. Downside is that it won't be explicitly visible in the config file. Hm I'll add options to prepend load-path dirs to avoid the ugly echo but to keep the information in the config file.